actor.go

```diff
@@ -28,89 +25,106 @@ type ActorConfig struct {
 // NewActor returns a new Actor
 func NewActor(c ActorConfig) (*Actor, error) {
 	if c.MailboxSize < 1 {
-		return nil, MailboxSizeTooSmallError("mailbox must be greater than 0")
+		return nil, ActorConfigurationError("mailbox must be greater than 0")
```

Comment:
This is a pattern i'm using in another project right now (where the text isn't always static), and I've gotten used to doing it. It's a little overkill, but i'd rather have a consistent means of providing typed errors than sometimes providing a type, and sometimes not.

I'm sure people will have opinions about it :)